components/apimgt/org.wso2.carbon.apimgt.tracing/src/main/java/org/wso2/carbon/apimgt/tracing/ZipkinTracer.java

```diff
@@ -0,0 +1,84 @@
+/*
+ * Copyright (c) 2018 WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+ *
+ * WSO2 Inc. licenses this file to you under the Apache License,
+ * Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.wso2.carbon.apimgt.tracing;
+
+import brave.Tracing;
+import brave.opentracing.BraveTracer;
+import brave.propagation.B3Propagation;
+import brave.propagation.ExtraFieldPropagation;
+import io.opentracing.Tracer;
+import io.opentracing.contrib.reporter.Reporter;
+import io.opentracing.contrib.reporter.TracerR;
+import io.opentracing.util.GlobalTracer;
+import io.opentracing.util.ThreadLocalScopeManager;
+import org.apache.commons.logging.LogFactory;
+import org.wso2.carbon.apimgt.impl.APIManagerConfiguration;
+import zipkin2.reporter.AsyncReporter;
+import zipkin2.reporter.okhttp3.OkHttpSender;
+
+/**
+ * Class for getting Zipkin tracer from reading configuration file
+ * */
+public class ZipkinTracer implements OpenTracer {
+
+    private static final String NAME = "zipkin";
+    private static APIManagerConfiguration configuration = new TracingServiceImpl().getConfiguration();
```

Comment:
Why the APIManagerConfiguration is loaded from TracingServiceImpl instance??

Reply:
Fixed. Loading APIManagerConfiguration from service reference holdeer